server/src/main/java/org/opensearch/extensions/ExtensionDependency.java

```diff
@@ -54,6 +57,25 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeVersion(version);
     }
 
+    public static ExtensionDependency parse(XContentParser parser) throws IOException {
+        String uniqueId = null;
+        Version version = null;
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+        while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+            String fieldName = parser.currentName();
+            parser.nextToken();
+
+            if ("uniqueId".equals(fieldName)) {
```

Comment:
Just a nit, let's use a switch statement in this parse method to handle parsing the different fields and add a default